### PR TITLE
UI Modeler cannot import reactivate event listener

### PR DESCRIPTION
**Pull request by aplatakis**
Issue:
It is not possible to import reactivate event listener in modeler UI. If cmmn diagram contains such element, then after import it is dropped and is not present in UI
Modeler UI not allows to add reactivate event listener using UI

Cause:
Reactivate event listener do not have json converter implemented.

Details:
Reactivate event listener is using the same tag "EventListener" as generic event listener. EventType="reactivate" is used to configure it as reactivate event listener

Solution:
* New stencil id introduced for reactivate listener
* For reactivate event listener to import registered new json converter which maps listener to "ReactivateEventListener" stencilId.
* For event registry support reactivation event listener can accept eventType. After this request will be accepted, other pull request will be posted for event registry support

#### Check List:
* Unit tests: NA
* Documentation: NA

### CODE REVIEW

**Comment by tijsrademakers** [comment]
Thanks for the PR, but for the reactivate event listener a dedicated element type would be needed, so not sharing it with a generic event listener. Would that be something you would be willing to work on?

**Comment by aplatakis** [comment]
@tijsrademakers thanks for review. First idea was to use new element type, but when I realized that I would need to introduce new icon for that, I dropped the idea. But I do agree that new element type would be better as this would drop not needed configuration options in UI, bring more clarity in UI and code and so on. I will change my code to use new element type
I have question regarding languages support. Is it enough to add localized values only for en? Will this not cause issues for other existing languages? As I'm not sure that google translate is the best option here.. For some languages I see that not all UI elements are translated

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/ReactivateEventListenerActivityBehaviour.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.flowable.cmmn.api.delegate.DelegatePlanItemInstance;
+import org.flowable.cmmn.api.runtime.PlanItemInstanceState;
+import org.flowable.cmmn.engine.impl.agenda.CmmnEngineAgenda;
+import org.flowable.cmmn.engine.impl.behavior.CmmnActivityBehavior;
+import org.flowable.cmmn.engine.impl.behavior.CoreCmmnTriggerableActivityBehavior;
+import org.flowable.cmmn.engine.impl.behavior.PlanItemActivityBehavior;
+import org.flowable.cmmn.engine.impl.persistence.entity.PlanItemInstanceEntity;
+import org.flowable.cmmn.engine.impl.util.CommandContextUtil;
+import org.flowable.cmmn.engine.impl.util.ExpressionUtil;
+import org.flowable.cmmn.engine.impl.util.PlanItemInstanceUtil;
```
[the remark this answers is not in the document]
Comment (aplatakis) [reply modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/ReactivateEventListenerActivityBehaviour.java]: Good catch. Thank you. Fixed

File: modules/flowable-cmmn-converter/src/main/java/org/flowable/cmmn/converter/export/ReactivationEventListenerExport.java [hunk 2]
```diff
@@ -48,6 +49,15 @@ protected void writePlanItemDefinitionSpecificAttributes(ReactivateEventListener
                 CmmnXmlConstants.ATTRIBUTE_EVENT_LISTENER_AVAILABLE_CONDITION,
                 reactivationEventListener.getAvailableConditionExpression());
         }
+
+        if (StringUtils.isNotEmpty(reactivationEventListener.getEventType()) && reactivationEventListener.getExtensionElements().get("eventType") == null) {
```
Comment (filiphr) [inline modules/flowable-cmmn-converter/src/main/java/org/flowable/cmmn/converter/export/ReactivationEventListenerExport.java:53]: Why do we need this change? there is already an exported flowable attribute with listener type `reactivate` on line 45. Perhaps the JSON conversion should use that instead.

**Review by filiphr — CHANGES_REQUESTED** [review]
thanks @aplatakis.

I see that the `ReactivateEventListener` is still extending from the `GenericEventListener`. That doesn't make sense. Can you please remove all the changes from the XML converter and CMMN engine and only have changes in the UI Modeler code. There, should be no need to change anything in the XML conversion and CMMN Engine for this functionality to work.

**Comment by tijsrademakers** [comment]
Closing because Modeler is not part of Flowable 7 codebase.

**Pull request closed without merging**